DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -101,16 +142,255 @@ void FillBindingData(
+        case DML_TENSOR_DATA_TYPE_INT32: return 4;
+        case DML_TENSOR_DATA_TYPE_INT16: return 2;
+        case DML_TENSOR_DATA_TYPE_INT8: return 1;
+        default: throw std::runtime_error("Unknown data type");
+    }
+}
+
+void DmlDispatchable::BuildGraph()
+{
+    const auto& desc = std::get<Model::DmlSerializedGraphDispatchableDesc>(m_desc);
+    
+    // Deserialize the graph
+    std::ifstream inFile(desc.sourcePath, std::ios::binary | std::ios::ate);
+    if (!inFile)
+    {
+        throw std::invalid_argument("Could not open the graph file for DmlSerializedGraph dispatchable");
+    }
+    std::streampos fileSize = inFile.tellg();
+    std::vector<uint8_t> blob(gsl::narrow_cast<size_t>(fileSize));
+    inFile.seekg(0, std::ios::beg);
+    inFile.read(reinterpret_cast<char*>(blob.data()), fileSize);
+
+    std::vector<std::unique_ptr<std::byte[]>> rawData;
+    DmlSerializedGraphDesc serializedDesc = DeserializeDmlGraph(blob.data(), rawData);
+
+    m_bindPoints = GetSerializedBindPoints(serializedDesc);
+    m_constantDataTypes = ExtractConstantDataTypes(serializedDesc);
+    m_serializedGraphBindings  = GenerateInitialBindingsFromGraph(serializedDesc);
+
+    for (const auto& node : serializedDesc.Nodes)
+    {
+        const auto* constantVariantPtr = std::get_if<DmlSerializedGraphNodeConstantVariant>(&node.Desc);
+        if (constantVariantPtr && std::holds_alternative<ConstantName>(*constantVariantPtr))
+        {
+            CreateResourceFromConstantNode(node);
+        }
+    }
+
+    // Convert to Public Graph Description
+    StackAllocator<1024> allocator;
+    DML_GRAPH_DESC dmlGraphDesc = {};
+    std::vector<Microsoft::WRL::ComPtr<IDMLOperator>> dmlOperators;
+    std::vector<DML_GRAPH_NODE_DESC> dmlGraphNodes;
+    std::vector<DML_GRAPH_EDGE_DESC> dmlInputEdges;
+    std::vector<DML_GRAPH_EDGE_DESC> dmlOutputEdges;
+    std::vector<DML_GRAPH_EDGE_DESC> dmlIntermediateEdges;
+
+    ConvertGraphDesc<1024>(
+        serializedDesc,
+        serializedDesc.InputCount,
+        serializedDesc.OutputCount,
+        m_device->DML(),
+        allocator,
+        nullptr, 
+        nullptr, 
+        dmlGraphDesc,
+        dmlOperators,
+        dmlGraphNodes,
+        dmlInputEdges,
+        dmlOutputEdges,
+        dmlIntermediateEdges);
+
+    //Compile the graph
+    THROW_IF_FAILED(m_device->DML()->CompileGraph(
+        &dmlGraphDesc,
+        DML_EXECUTION_FLAG_NONE,
+        IID_PPV_ARGS(&m_compiledOperator)));
+}
+
 void DmlDispatchable::Initialize()
 {
-    THROW_IF_FAILED(m_device->DML()->CompileOperator(
-        m_operator.Get(), 
-        m_desc.executionFlags, 
-        IID_PPV_ARGS(m_operatorCompiled.ReleaseAndGetAddressOf())));
-    m_operatorCompiled->SetName(std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(m_name).data());
+   if (std::holds_alternative<Model::DmlDispatchableDesc>(m_desc))
+    {
+        CompileOperator();
+    }
+    else
+    {
+        BuildGraph();
```

Comment:
nit: should we rename the method to `BuildAndCompileGraph` to make it clearer that after this method we will also have the graph compiled.